horreum-backend/src/main/java/io/hyperfoil/tools/horreum/bus/MessageBus.java

```diff
@@ -248,36 +248,43 @@ void removeIndex(String channel, int index) {
          delayed = "{horreum.messagebus.retry.delay:30s}",
          concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
    public void retryFailedMessages() {
-      Query query = em.createNativeQuery("SELECT channel, id, testid, flags, message FROM messagebus WHERE \"timestamp\" + make_interval(secs => ?1) <= now()")
+
+      try(ScrollableResults<ChannelMessage> results = (ScrollableResults<ChannelMessage>) em.createNativeQuery("SELECT channel, id, testid, flags, message FROM messagebus WHERE \"timestamp\" + make_interval(secs => ?1) <= now()")
             .setParameter(1, retryAfter.toSeconds())
             .unwrap(NativeQuery.class)
-            .addScalar("channel", TextType.INSTANCE)
-            .addScalar("id", LongType.INSTANCE)
-            .addScalar("testid", IntegerType.INSTANCE)
-            .addScalar("flags", IntegerType.INSTANCE)
-            .addScalar("message", JsonNodeBinaryType.INSTANCE);
-      try (ScrollableResults results = Util.scroll(query)) {
+            .addScalar("channel", StandardBasicTypes.TEXT)
+            .addScalar("id", StandardBasicTypes.LONG)
+            .addScalar("testid", StandardBasicTypes.INTEGER)
+            .addScalar("flags", StandardBasicTypes.INTEGER)
+            .addScalar("message", new CustomType<>(new JsonBinaryType(), new TypeConfiguration()))
+              .setTupleTransformer((tuples, aliases) -> {
```

Comment:
could we define an optional `tupleTransformer` param for `Util.scroll` and pass in the tuple lambda to remove duplicated code?